streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidBeamFactoryImpl.java

```diff
@@ -352,10 +356,10 @@ public StreamlineTimestamper(String timestampField) {
 
         @Override
         public DateTime timestamp(Map<String, Object> theMap) {
-            if (PROCESSING_TIME.equalsIgnoreCase(timestampField))
-                return new DateTime(System.currentTimeMillis());
+            if (!theMap.containsKey(timestampField))
```

Comment:
nit: can use the return value of get to avoid looking up the map twice (here and below)

Reply:
Thanks for the review. Updated the code